feat: Add mentionRaidProtectionEnabled

Harbinger-of-Death opened the pull request:
- This adds a settable trigger metadata property called mentionRaidProtectionEnabled.
- Also adds separate methods for setting trigger metadatas

Harbinger-of-Death committed:
feat: Add mentionRaidProtectionEnabled

- This adds a settable trigger metadata property called mentionRaidProtectionEnabled.
- Also adds separate methods for setting trigger metadatas

MrMythicalYT commented on Structures/AutoModeration.js:
```diff
@@ -16,14 +17,33 @@ class AutoModeration extends Base {
 
+    get exemptChannels() {
+        const collection = new Collection()
+        for(const snowflake of this._data.exempt_channels) {
+            if(!this.client.channels.cache.has(snowflake)) continue;
+            collection.set(snowflake, this.client.channels.cache.get(snowflake))
```
```suggestion
        for(const channel of this._data.exempt_channels) {
            if(!this.client.channels.cache.has(channel)) continue;
            collection.set(channel, this.client.channels.cache.get(channel))
```

MrMythicalYT commented on Structures/AutoModeration.js:
```diff
@@ -16,14 +17,33 @@ class AutoModeration extends Base {
+
+    get exemptRoles() {
+        const collection = new Collection()
+        for(const snowflake of this._data.exempt_roles) {
+            if(!this.guild?.roles.cache.has(snowflake)) continue;
+            collection.set(snowflake, this.guild?.roles.cache.get(snowflake))
```
```suggestion
        for(const role of this._data.exempt_roles) {
            if(!this.guild?.roles.cache.has(role)) continue;
            collection.set(role, this.guild?.roles.cache.get(role))
```

MrMythicalYT reviewed (changes requested):
More descriptive variable names

Harbinger-of-Death committed:
chore: Added descriptive namings

Harbinger-of-Death committed:
fix: Fixed ts docs

Harbinger-of-Death committed:
feat: Added methods for Automoderation

- These methods are shortcuts for modifying or adding metadatas to Automoderation

Harbinger-of-Death commented on index.d.ts:
```diff
@@ -654,6 +654,10 @@ export class AutoModeration extends Base {
+    /**
+     * Whether or not this Automod has mention protection enabled against raid. For MentionSpam trigger type
+     */
+    public mentionRaidProtectionEnabled: boolean
```
New undocumented property

Harbinger-of-Death replied:
don't tell discord

Harbinger-of-Death committed:
Merge branch 'main' into Add-Automoderation-Mention-Raid-Protection-Enabled

Harbinger-of-Death committed:
fix: Merged Conflict

Harbinger-of-Death committed:
fix: Merged Conflict

Harbinger-of-Death committed:
chore: Added mentionRaidProtectionEnabled

- This checks if mentionRaidProtectionEnabled is equals another rule mentionRaidProtectionEnabled

Harbinger-of-Death committed:
chore: Fixed a misplaced whitespace
